### ISSUE TITLE
valve_state seems mixed up

### ISSUE THREAD

**Issue by WallyDW** [SukramJ/hahomematic#1073]
### I agree to the following

- [X] I have read the [documentation](https://github.com/danielperna84/custom_homematic#custom_homematic)
- [X] I am aware of the latest [release notes](https://github.com/danielperna84/custom_homematic/releases)
- [X] The Backend (CCU/Homegear/...) is working as expected and connected devices are controllable by its own UI.
- [X] I am running the latest version of the custom_component (and Home Assistant)

### The problem

The indication of a conversion to float and the measurement unit set to % indicates that the valve_state is thought to be the percentage of valve opening.
The paramset VALVE_STATE is a numeric value from 0 to 4 indicating the state of the physical mounting of the thermostat to the valve, which does not fit with the value trying to be converted, because that is a string (adaption_done).
adaption_done indicates that the value is actually from the paramset VALVE_ADAPTION
The percentage of valve opening is the paramset LEVEL

### What version of HomematicIP (local) has the issue?

1.37.1

### What type of installation are you running?

Home Assistant Supervised

### What type of installation are you running for your homematic backend?

Debmatic

### Which version of your homematic backend are you running?

3.69.7.105

### What hardware are you running for your system?

X64

### Which config details do you use

- [X] Username and password
- [ ] TLS
- [ ] callback data ([see](https://github.com/danielperna84/custom_homematic#callback_host-and-callback_port))

### Which interfaces do you use?

- [X] Homematic IP
- [ ] Homematic (Bidcos-RF)
- [ ] Groups (Heating-Group)
- [ ] BidCos-Wired (HM-Wired)

### Anything in the logs that might be useful for us?

```txt
2023-06-02 05:37:58.762 ERROR (MainThread) [homeassistant.components.sensor] Error adding entities for domain sensor with platform homematicip_local
Traceback (most recent call last):
File "/usr/src/homeassistant/homeassistant/components/sensor/__init__.py", line 581, in state
numerical_value = float(value) # type:ignore[arg-type]
ValueError: could not convert string to float: 'adaption_done'
The above exception was the direct cause of the following exception:
Traceback (most recent call last):
File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 455, in async_add_entities
await asyncio.gather(*tasks)
File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 731, in _async_add_entity
await entity.add_to_platform_finish()
File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 844, in add_to_platform_finish
await self.async_added_to_hass()
File "/config/custom_components/homematicip_local/sensor.py", line 200, in async_added_to_hass
await super().async_added_to_hass()
File "/config/custom_components/homematicip_local/generic_entity.py", line 175, in async_added_to_hass
await self._hm_entity.load_entity_value(call_source=HmCallSource.HA_INIT)
File "/usr/local/lib/python3.10/site-packages/hahomematic/platforms/entity.py", line 460, in load_entity_value
self.update_value(
File "/usr/local/lib/python3.10/site-packages/hahomematic/platforms/entity.py", line 479, in update_value
self.update_entity()
File "/usr/local/lib/python3.10/site-packages/hahomematic/platforms/entity.py", line 236, in update_entity
super().update_entity(*args)
File "/usr/local/lib/python3.10/site-packages/hahomematic/platforms/entity.py", line 144, in update_entity
_callback(*args)
File "/config/custom_components/homematicip_local/generic_entity.py", line 195, in _async_device_changed
self.async_write_ha_state()
File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 585, in async_write_ha_state
self._async_write_ha_state()
File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 649, in _async_write_ha_state
state = self._stringify_state(available)
File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 591, in _stringify_state
if (state := self.state) is None:
File "/usr/src/homeassistant/homeassistant/components/sensor/__init__.py", line 583, in state
raise ValueError(
ValueError: Sensor sensor.living_room_thermostat_valve_state has device class None, state class measurement unit % and suggested precision None thus indicating it has a numeric value; however, it has the non-numeric value: adaption_done (<class 'str'>)
```

### Additional information

Thermostat's output in XMLAPI.
```
<channel name="Living Room Thermostat:1" ise_id="2123" index="1" visible="true" operate="true">
<datapoint name="HmIP-RF.00119A499C1379:1.ACTIVE_PROFILE" type="ACTIVE_PROFILE" ise_id="2124" value="1" valuetype="16" valueunit="" timestamp="1685671087" operations="7"/>
<datapoint name="HmIP-RF.00119A499C1379:1.ACTUAL_TEMPERATURE" type="ACTUAL_TEMPERATURE" ise_id="2125" value="24.200000" valuetype="4" valueunit="°C" timestamp="1685671087" operations="5"/>
<datapoint name="HmIP-RF.00119A499C1379:1.ACTUAL_TEMPERATURE_STATUS" type="ACTUAL_TEMPERATURE_STATUS" ise_id="2126" value="0" valuetype="16" valueunit="" timestamp="1685671087" operations="5"/>
<datapoint name="HmIP-RF.00119A499C1379:1.BOOST_MODE" type="BOOST_MODE" ise_id="2127" value="false" valuetype="2" valueunit="" timestamp="1685671087" operations="6"/>
<datapoint name="HmIP-RF.00119A499C1379:1.BOOST_TIME" type="BOOST_TIME" ise_id="2128" value="0" valuetype="16" valueunit="" timestamp="1685671087" operations="5"/>
<datapoint name="HmIP-RF.00119A499C1379:1.CONTROL_DIFFERENTIAL_TEMPERATURE" type="CONTROL_DIFFERENTIAL_TEMPERATURE" ise_id="2129" value="" valuetype="4" valueunit="" timestamp="0" operations="2"/>
<datapoint name="HmIP-RF.00119A499C1379:1.CONTROL_MODE" type="CONTROL_MODE" ise_id="2130" value="" valuetype="16" valueunit="" timestamp="0" operations="2"/>
<datapoint name="HmIP-RF.00119A499C1379:1.DURATION_UNIT" type="DURATION_UNIT" ise_id="2131" value="" valuetype="16" valueunit="" timestamp="0" operations="2"/>
<datapoint name="HmIP-RF.00119A499C1379:1.DURATION_VALUE" type="DURATION_VALUE" ise_id="2132" value="" valuetype="16" valueunit="" timestamp="0" operations="2"/>
<datapoint name="HmIP-RF.00119A499C1379:1.FROST_PROTECTION" type="FROST_PROTECTION" ise_id="2133" value="false" valuetype="2" valueunit="" timestamp="1685671087" operations="5"/>
<datapoint name="HmIP-RF.00119A499C1379:1.LEVEL" type="LEVEL" ise_id="2134" value="0.000000" valuetype="4" valueunit="" timestamp="1685671087" operations="7"/>
<datapoint name="HmIP-RF.00119A499C1379:1.LEVEL_STATUS" type="LEVEL_STATUS" ise_id="2135" value="0" valuetype="16" valueunit="" timestamp="1685671087" operations="5"/>
<datapoint name="HmIP-RF.00119A499C1379:1.PARTY_MODE" type="PARTY_MODE" ise_id="2136" value="false" valuetype="2" valueunit="" timestamp="1685671087" operations="5"/>
<datapoint name="HmIP-RF.00119A499C1379:1.PARTY_SET_POINT_TEMPERATURE" type="PARTY_SET_POINT_TEMPERATURE" ise_id="2137" value="0.000000" valuetype="4" valueunit="" timestamp="0" operations="5"/>
<datapoint name="HmIP-RF.00119A499C1379:1.PARTY_TIME_END" type="PARTY_TIME_END" ise_id="2138" value="" valuetype="20" valueunit="" timestamp="0" operations="7"/>
<datapoint name="HmIP-RF.00119A499C1379:1.PARTY_TIME_START" type="PARTY_TIME_START" ise_id="2139" value="" valuetype="20" valueunit="" timestamp="0" operations="7"/>
<datapoint name="HmIP-RF.00119A499C1379:1.QUICK_VETO_TIME" type="QUICK_VETO_TIME" ise_id="2140" value="0" valuetype="16" valueunit="" timestamp="1685671087" operations="5"/>
<datapoint name="HmIP-RF.00119A499C1379:1.SET_POINT_MODE" type="SET_POINT_MODE" ise_id="2141" value="1" valuetype="16" valueunit="" timestamp="1685671087" operations="7"/>
<datapoint name="HmIP-RF.00119A499C1379:1.SET_POINT_TEMPERATURE" type="SET_POINT_TEMPERATURE" ise_id="2142" value="19.500000" valuetype="4" valueunit="°C" timestamp="1685671087" operations="7"/>
<datapoint name="HmIP-RF.00119A499C1379:1.SWITCH_POINT_OCCURED" type="SWITCH_POINT_OCCURED" ise_id="2143" value="false" valuetype="2" valueunit="" timestamp="1685671087" operations="5"/>
<datapoint name="HmIP-RF.00119A499C1379:1.VALVE_ADAPTION" type="VALVE_ADAPTION" ise_id="2144" value="false" valuetype="2" valueunit="" timestamp="0" operations="7"/>
<datapoint name="HmIP-RF.00119A499C1379:1.VALVE_STATE" type="VALVE_STATE" ise_id="2145" value="4" valuetype="16" valueunit="" timestamp="1685671087" operations="5"/>
<datapoint name="HmIP-RF.00119A499C1379:1.WINDOW_STATE" type="WINDOW_STATE" ise_id="2146" value="0" valuetype="16" valueunit="" timestamp="1685671087" operations="7"/>
</channel>
```

**Comment by SukramJ** [comment]
Which device are you talking about?
Did you add valve_state to unignore?
Which HA version do you use?

**Comment by SukramJ** [comment]
VALVE_STATE is a parameter that also occurs with numeric values in HM, and in this case it is directly supported by the integration. 
If additional parameters are added via unignore, you do this at your **own risk**, as also described in the unignore instructions!

**Comment by SukramJ** [comment]
Fixed with next release 1.37.2.
